server/src/main/java/org/apache/iotdb/db/service/InternalServiceImpl.java

```diff
@@ -43,39 +45,73 @@
 import org.apache.iotdb.rpc.TSStatusCode;
 
 import org.apache.thrift.TException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
 
 public class InternalServiceImpl implements InternalService.Iface {
+  private static final Logger LOGGER = LoggerFactory.getLogger(InternalServiceImpl.class);
+
+  private final ConsensusManager consensusManager;
 
-  public InternalServiceImpl() {
+  public InternalServiceImpl() throws IOException {
     super();
+    consensusManager = new ConsensusManager();
   }
 
   @Override
-  public TSendFragmentInstanceResp sendFragmentInstance(TSendFragmentInstanceReq req)
-      throws TException {
+  public TSendFragmentInstanceResp sendFragmentInstance(TSendFragmentInstanceReq req) {
+    TSendFragmentInstanceResp response = new TSendFragmentInstanceResp();
+    FragmentInstance fragmentInstance = null;
+    try {
+      fragmentInstance = FragmentInstance.deserializeFrom(req.fragmentInstance.body);
```

Comment:
For forwarded requests, we tend to move the deserialization logic below the consensus layer